examples/loki/nurf_data.py

```diff
@@ -107,115 +107,163 @@ def nurf_file_creator(loki_file, path_to_loki_file, data):
         #print(list(hf['/entry/'].keys()))
         
         # create the various subgrous and their attributes
-
+        
         # UV subgroup
         grp_uv = hf.create_group("/entry/instrument/uv")
-        grp_uv.attrs["NX_class"] = 'NXdetector'
-        grp_uv.create_dataset('description', data='UV')
-        grp_uv.create_dataset('type',
-                              data='HR4PRO-UV-VIS-ES, DH-2000-FHS-DUV-TTL, QP600-2-SR')
+        grp_uv.attrs["NX_class"] = 'NXdetector_group'
+        
+        # subgroup for uv_spectra
+        uv_spectra=grp_uv.create_group("uv_spectra")
+        uv_spectra.attrs["NX_class"] = 'NXdetector'
+        uv_spectra_data=uv_spectra.create_dataset('data', data=data['UV_spectra'],
+                                           shape=data[
+                                               'UV_spectra'].shape,
+                                           dtype=np.int32)
```

Comment:
Is there a reason for giving `shape` and `dtype` explicitly? If they are omitted I think the simple get determined automatically?

Reply:
Re first comment: That is my problem with the 'NXdetector' class. Overall, it does not describe well these type of detectors. It does not have a field/dataset for wavelength for example, but the `uv_spectra` itself is data from a real detector. "Those should be fields within the other NXdetector groups." 
The previous version had one group uv as 'NXdetector' with subdatasets for wavelength, time, background, intensity0 etc. But technically, this does not fit well the definition of the 'NXdetector' class.

I thought I have to better define `shape` and `dtype`, so I left this as spaceholders.

Reply:
> That is my problem with the 'NXdetector' class. Overall, it does not describe well these type of detectors. It does not have a field/dataset for wavelength for example, but the uv_spectra itself is data from a real detector. "Those should be fields within the other NXdetector groups."
The previous version had one group uv as 'NXdetector' with subdatasets for wavelength, time, background, intensity0 etc. But technically, this does not fit well the definition of the 'NXdetector' class.

@zjttoefs may be able to answer this better.

> I thought I have to better define shape and dtype, so I left this as spaceholders.

Unless you want to force, e.g., the `dtype` to something other than the dtype of the numpy array you are storing, I don't think you have to?